README.md

```diff
@@ -2,19 +2,26 @@
 
 In a number of different settings, interactions between clients and servers
 involve information that could be sensitive when associated with client
-identity.
-
-Client-server protocols like HTTP reveal aspects of client identity to servers
-through these interactions, especially source addresses.  Even without client
-identity, a server might be able to build a profile of client activity by
-correlating requests from the same client over time.
-
-In a setting where the information included in requests does not need to be
-correlated, the Oblivious HTTP protocol allows a server to accept requests via a
-proxy.  The proxy ensures that the server cannot see source addressing
+identity.  Client-server applications built on HTTP reveal aspects of client
+identity to servers through these interactions, especially source addresses.
+Even without client identity, a server might be able to build a profile of
+client activity by correlating requests from the same client over time.
+
+In HTTP-based applications where the information included in requests does not
+need to be correlated, the Oblivious HTTP protocol allows a supporting server to accept
+requests via a proxy.  The proxy ensures that the server cannot see source addressing
 information for clients, which prevents servers linking requests to the same
-client.  Encryption ensures that the proxy is unable to read requests or
-responses.
+client using such information.  Encryption ensures that the proxy is unable to read
+requests or responses.
+
+Example applications and use cases fit for the Oblivious HTTP protocol include DNS,
+data or telemetry submission, and special-purpose API calls such as Safe Browsing.
+In such application deployments, the relationship between client, server, and
+cooperating proxy is typically configured out-of-band.  General purpose HTTP
+applications such as web browsing are not in scope for the Oblivious HTTP protocol,
+which requires explicit new server support. Transport-level proxies such as
```

Comment:
This "explicit support" tail seems tacked on.  There are two good reasons for not using this for browsing (probably more): browsing generally relies extensively on linking requests and broader applicability is limited by the need for explicit support of the new protocol.

Reply:
How about the following:

```
General purpose HTTP applications such as web browsing are not in scope for the Oblivious HTTP protocol.
Broad applicability is limited by the need for explicit server support of the protocol. In contrast, transport-level...
```

Reply:
WFM

Reply:
If the charter is going to list one and only one reason for out-of-scoping web browsing, I would suggest the other one, that web browsing generally relies on linkability and thus the security properties of Oblivious are not desired for that case.  The problem with listing just the "explicit support" argument is that if that were the only issue, it's not hard to come up with technical solutions to the issue (and is already solved in the current draft through the "request resource"/"target resource" split).

The "explicit support" argument is more just a simple argument to not worry about it, not the main reason to not try to do it and declare it out of scope.

Reply:
I noted that explicit support was just one of the reasons broad applicability isn't the end goal, which I think addresses your concern @ericorth. Can you confirm?

Reply:
My point is that I object to the "Broad applicability is limited by the need for explicit server support of the protocol." sentence in your last comment because it's unclear and unhelpful for that to be the sole reason listed.  I would be happier with:

- "The security properties of the protocol are not useful for general purpose HTTP applications."
- "Broad applicability is limited by the need for explicit server support of the protocol, and the security properties of the protocol are not useful for general purpose HTTP applications."
- <Nothing.  Just let the "General purpose [...] are not in scope [...]" sentence stand on its own.>

Reply:
"Broad applicability is limited by the need for explicit server support of the protocol, and the security properties of the protocol are not useful for general purpose HTTP applications" --> is the same, to me, as what's there currently.

Reply:
I see...

> Broad applicability is limited by multiple factors, including the need for explicit server support of the protocol.

Is github hiding an update from me? Still only lists one reason, and is IMO, the less helpful reason.

Reply:
Nope, that's the right text. It doesn't attempt to enumerate all the reasons why you don't want this for general purpose web browsing (including linkability and so on). It only lists one.

Reply:
Got it.

I suppose the "limited by multiple factors" part is better than nothing at resolving my concern, and I'll live with it if I'm the only person complaining about the selection of the reason.  But I still believe the one listed reason is not the more important one and not the most helpful one to list if we're just going to list one.

Reply:
Well, you can't do OHTTP without server-side support, so that seems to simply take precedence over everything else 🤷 This seems like a bike shed though, so I'll let someone else who has stronger feelings resolve it.

Reply:
> Well, you can't do OHTTP without server-side support

You can though, and it's not that hard.  All you have to do is route it to one server that unpackages the oblivious message and then passes it on to an HTTP server without specific support.  The most recent oHTTP and ODoH drafts both explicitly allowed this behavior.  I feel like by putting the explicit support argument in the charter, this is only going to lead to a future argument when discussing whether or not to allow such unpackaging behavior.